### PR TITLE
chore(Jest): SearchParameters [PART-2]

### PR DESCRIPTION
**Pull request by samouss**
This PR moves the folder `test/spec/SearchParameters` to Jest. You can run the tests locally with:

```
yarn jest test/spec/SearchParameters
```

The migration has been done by [`jest-codemods`](https://github.com/skovhus/jest-codemods) (recommended by the [Jest](https://jestjs.io/docs/en/migration-guide#jest-codemods) project) for all the assertions. Then I removed the dependency on `sinon` file by file. I tried to didn't update the code style or anything unless it was too complicated to migrate.

### CODE REVIEW

File: test/spec/SearchParameters/isNumericRefined.js [hunk 1]
```diff
@@ -1,59 +1,52 @@
...
 
-test('isNumericRefined with 3 parameters', function(t) {
+test('isNumericRefined with 3 parameters', function() {
   var params = new SearchParameters();
 
-  t.notOk(params.isNumericRefined('age', '>', '3'));
-  t.notOk(params.isNumericRefined('age', '>', '7'));
-  t.notOk(params.isNumericRefined('age', '<', '7'));
-  t.notOk(params.isNumericRefined('size', '>', '3'));
+  expect(params.isNumericRefined('age', '>', '3')).toBeFalsy();
```
Comment (francoischalifour) [inline test/spec/SearchParameters/isNumericRefined.js]: `isNumericRefined` returns a boolean so we can make the assertion more strict with:

```js
expect(params.isNumericRefined('age', '>', '3')).toBe(false);
```

File: test/spec/SearchParameters/isNumericRefined.js [hunk 2]
```diff
@@ -1,59 +1,52 @@
...
+  expect(params.isNumericRefined('size', '>', '3')).toBeFalsy();
 
   var paramsWithNumerics = (params.addNumericRefinement('age', '>', '3'));
 
-  t.ok(paramsWithNumerics.isNumericRefined('age', '>', '3'));
-  t.ok(paramsWithNumerics.isNumericRefined('age', '>', 3));
-  t.notOk(paramsWithNumerics.isNumericRefined('age', '>', '7'));
-  t.notOk(paramsWithNumerics.isNumericRefined('age', '<', '7'));
-  t.notOk(paramsWithNumerics.isNumericRefined('size', '>', '3'));
+  expect(paramsWithNumerics.isNumericRefined('age', '>', '3')).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/SearchParameters/isNumericRefined.js]: `isNumericRefined` returns a boolean so we can make the assertion more strict.

File: test/spec/SearchParameters/listAttributes.js [hunk 3]
```diff
@@ -1,88 +1,72 @@
...
-  t.throws(function() {
+test('addHierarchicalFacet should throw when a facet with the same name is already declared', function() {
+  expect(function() {
     SearchParameters
       .make({hierarchicalFacets: [{name: 'facet'}]})
       .addHierarchicalFacet({name: 'facet'});
-  });
-
-  t.end();
+  }).toThrow();
```
Comment (francoischalifour) [inline test/spec/SearchParameters/listAttributes.js]: Can we use an inline snapshot now that we use Jest?

File: test/spec/SearchParameters/_parseNumbers.js [hunk 4]
```diff
@@ -20,56 +19,48 @@ test('_parseNumbers should convert to number all specified root keys (that are p
...
+test('_parseNumbers should not convert undefined to NaN', function() {
   var partialState = {
     aroundPrecision: undefined
   };
   var actual = SearchParameters._parseNumbers(partialState);
 
-  t.equal(actual.aroundPrecision, undefined);
-
-  t.end();
+  expect(actual.aroundPrecision).toBe(undefined);
```
Comment (francoischalifour) [inline test/spec/SearchParameters/_parseNumbers.js]: We can use [`toBeUndefined()`](https://jestjs.io/docs/en/expect#tobeundefined).

File: test/spec/SearchParameters/isNumericRefined.js (hunk 1 above)
Comment (samouss) [reply test/spec/SearchParameters/isNumericRefined.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/SearchParameters/isNumericRefined.js (hunk 2 above)
Comment (samouss) [reply test/spec/SearchParameters/isNumericRefined.js]: https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/SearchParameters/listAttributes.js (hunk 3 above)
Comment (samouss) [reply test/spec/SearchParameters/listAttributes.js]: https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278627422

File: test/spec/SearchParameters/_parseNumbers.js (hunk 4 above)
Comment (samouss) [reply test/spec/SearchParameters/_parseNumbers.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

**Pull request merged**